### PR TITLE
Update `nodejs-engine` to `libcnb@0.22` API

### PR DESCRIPTION
**Pull request by colincasey**
This builds on top of #877 to remove the `#allow(deprecated)` annotations for the `nodejs-engine` CNB. No new functionality has been added. The trait layer code has simply been moved from:
- `layers/dist.rs` → `install_node.rs`
- `layers/node_runtime_metrics.rs` → `attach_runtime_metrics.rs`
- `layers/web_env.rs` → `configure_web_env.rs`

### CODE REVIEW

**Review by edmorley — COMMENTED** [review]
CI is failing

File: buildpacks/nodejs-function-invoker/src/main.rs [hunk 1]
```diff
@@ -69,6 +69,7 @@ impl Buildpack for NodeJsInvokerBuildpack {
             .unwrap_or_else(|| DetectResultBuilder::fail().build())
     }
 
+    #[allow(deprecated)]
```
Comment (edmorley) [inline buildpacks/nodejs-function-invoker/src/main.rs:72]: Is this addition intentional, or a rebase issue?

```suggestion
```

File: buildpacks/nodejs-corepack/src/main.rs [hunk 2]
```diff
@@ -63,6 +63,7 @@ impl Buildpack for CorepackBuildpack {
         })
     }
 
+    #[allow(deprecated)]
```
Comment (edmorley) [inline buildpacks/nodejs-corepack/src/main.rs:66]: Is this addition intentional, or a rebase issue?

```suggestion
```

**Review by edmorley — APPROVED** [review]

File: buildpacks/nodejs-function-invoker/src/main.rs (hunk 1 above)
Comment (colincasey) [reply buildpacks/nodejs-function-invoker/src/main.rs]: Rebase issue that, for whatever reason, only happened with this PR and none of the other. This `#[allow(deprecated)]` annotation is now removed (again).

File: buildpacks/nodejs-corepack/src/main.rs (hunk 2 above)
Comment (colincasey) [reply buildpacks/nodejs-corepack/src/main.rs]: This `#[allow(deprecated)]` annotation is now removed (again).
